homeassistant/components/gotify/__init__.py

```diff
@@ -0,0 +1,24 @@
+"""The gotify component."""
+import voluptuous as vol
+
+from homeassistant.const import CONF_TOKEN, CONF_URL
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {vol.Required(CONF_TOKEN): cv.string, vol.Required(CONF_URL): cv.string}
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
```

Comment:
New integrations that communicate with devices and/or services should be configurable with a Config Flow instead of YAML.

For more detail read [ADR-0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision)

Reply:
Ok, I will look at how to set this up.

Reply:
Hi, I've attempted to migrate this new component over to a Config Flow. This is new to me so I have pushed these changes onto a new branch and would like someone to informally review and provide and guidance/suggestions if things need improved before I merge those changes onto this branch for the PR. The new component has been tested locally and functionally works, the config flow is tested and also seems to work fine. @milanmeu could you please take a quick look at [these changes](https://github.com/benjmarshall/core/commit/4532ed1450db94c2506ac52e66034283239deddb) and provide any advice you can or let me know if you think it's ready to merge into this PR?

Reply:
I tested this locally as well and it works for me 👍

Reply:
> @milanmeu could you please take a quick look at [these changes](https://github.com/benjmarshall/core/commit/4532ed1450db94c2506ac52e66034283239deddb) and provide any advice you can or let me know if you think it's ready to merge into this PR?

Here is my comment: https://github.com/benjmarshall/core/pull/5.

Merge `gotify_config_flow` into `add_gotify_integration` so other users/reviewers can see what's happening.